crates/telio-nurse/src/aggregator.rs

```diff
@@ -322,11 +330,18 @@ impl<W: WireGuard> ConnectivityDataAggregator<W> {
         }
     }
 
+    #[allow(dead_code)] //TODO(Hasan) : Remove this!
+    pub async fn clear_segments(&self) {
+        let mut data_guard = self.data.lock().await;
+        data_guard.current_peer_events.clear();
+        data_guard.current_relay_event.take();
+    }
+
```

Comment:
This is like a failsafe in case of race conditions where we get the notification to sleep, but before system actually goes to sleep there is a change in state and for the whole duration of the sleep that state is taken instead.

Reply:
> If you call the other function on sleep, this shouldn't be needed probably?


Yeah, to add a little bit more, we were discussing about situation like this:
```
  |-------------------------xxxxx| zzzzzzzzzzzz |xxxx------------------------------|
start                         ^                                        ^
                       sleep notification
```
The sleep and wake notification necessarily have to be called some time before actual sleep, and wakeup notification some time after actual sleep. Therefore there are these periods (I marked them with `x`), which is this "dangerous territory" where an event change would produce incorrect events.

While clearing at the wakeup notification is not ideal situation -> it is better than nothing.